pkg/component/extensions/worker/worker.go

```diff
@@ -262,10 +262,28 @@ func (w *worker) deploy(ctx context.Context, operation string) (extensionsv1alph
 
 // Restore uses the seed client and the ShootState to create the Worker resources and restore their state.
 func (w *worker) Restore(ctx context.Context, shootState *gardencorev1beta1.ShootState) error {
+	// gardenlet persists the machine state in the ShootState's `.spec.gardener[]` list with `type=machine-state`.
+	// In order to use below function, we have to make the worker state appear under `.spec.extensions[]` list with
+	// the correct kind and name.
+	var (
+		shootStateCopy = shootState.DeepCopy()
```

Comment:
Can you explain why we need a copy here?

Reply:
A few lines below, I'm adding something to the `ShootState` which should only be visible in this function, i.e., I don't want to modify the passed object.